system-tests/mpi/osu/osu_check.py

```diff
@@ -0,0 +1,179 @@
+class lumi_osu_pt2pt_check(lumi_osu_benchmarks):
+    valid_systems = ['lumi:small', 'lumi:gpu'] 
+    use_multithreading = False
+    time_limit = '10m'
+    benchmark_info = parameter([
+        ('mpi.pt2pt.osu_bw', 'bandwidth'),
+        ('mpi.pt2pt.osu_latency', 'latency')
+    ], fmt=lambda x: x[0], loggable=True)
+    osu_binaries = fixture(lumi_build_osu_benchmarks, scope='environment')
+    allref = {
+        'mpi.pt2pt.osu_bw': {
+            'cpu': {
+                'lumi:small': {
+                    'bandwidth': (6757.0, -0.10, None, 'MB/s')
+                }
+            },
+            'rocm': {
+                'lumi:gpu': {
+                    'bandwidth': (4850.0, -0.10, None, 'MB/s')
+                }
+            }
+        },
+        'mpi.pt2pt.osu_latency': {
+            'cpu': {
+                'lumi:small': {
+                    'latency': (1.8, None, 0.80, 'us')
+                }
+            },
+            'rocm': {
+                'lumi:gpu': {
+                    'latency': (3.75, None, 0.50, 'us')
+                }
+            }
+        }
+    }
+
+    @run_after('init')
+    def setup_per_build_type(self):
+        build_type = self.osu_binaries.build_type
+        if build_type == 'rocm':
+            self.valid_systems = ['lumi:gpu']
+            self.device_buffers = 'rocm'
+            self.num_gpus_per_node = 1
+            self.executable_opts = ['-c', '-d', 'rocm', 'D', 'D']
+            self.valid_prog_environs = ['builtin-hip']
+            self.variables = {'MPICH_GPU_SUPPORT_ENABLED': '1'} 
+        else:
+            self.valid_systems = ['lumi:small']
+            self.valid_prog_environs = ['cpeGNU', 'cpeCray']
+            self.executable_opts = ['-c']
+
+        with contextlib.suppress(KeyError):
+            self.reference = self.allref[self.benchmark_info[0]][build_type]
+
+    @run_before('run')
+    def add_exec_prefix(self):
+        build_type = self.osu_binaries.build_type
+        bench_path = self.benchmark_info[0].replace('.', '/')
+        # update directory structure
+        self.executable = os.path.join(self.osu_binaries.stagedir,
+                                       self.osu_binaries.build_prefix,
+                                       'c',
+                                       bench_path)
+        if build_type == 'rocm':
+            self.executable = os.path.join(self.osu_binaries.stagedir, self.osu_binaries.build_prefix,
+                                           'map_rank_to_gpu ') + self.executable
+
+    @sanity_function
+    def validate_test(self):
+        # with validation
+        return sn.assert_found(rf'^{self.message_size}.*Pass', self.stdout)
+
+
+@rfm.simple_test
+class lumi_osu_collective_check(lumi_osu_benchmarks):
+    benchmark_info = parameter([
+        #('mpi.collective.osu_alltoall', 'latency'),
+        ('mpi.collective.osu_allreduce', 'latency'),
+    ], fmt=lambda x: x[0], loggable=True)
+    num_nodes = parameter([1, 2])
```

Comment:
What about running it for a larger number of nodes? 8 o 16 maybe?

Reply:
4 is max for `small` partition, I am staying with the limit for now